cranelift/filetests/filetests/isa/x64/simd-bitwise-avx.clif

```diff
@@ -0,0 +1,128 @@
+test compile precise-output
+set enable_simd
+target x86_64 has_avx
+
+function %mask_from_icmp(f32x4, f32x4) -> f32x4 {
+block0(v0: f32x4, v1: f32x4):
+    v2 = fmin v0, v1
+    return v2
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   vminps  %xmm0, %xmm1, %xmm3
+;   vminps  %xmm1, %xmm0, %xmm5
+;   vorps   %xmm3, %xmm5, %xmm7
+;   vcmpps  $3 %xmm7, %xmm5, %xmm9
+;   vorps   %xmm7, %xmm9, %xmm11
+;   vpsrld  %xmm9, $10, %xmm13
+;   vandnps %xmm13, %xmm11, %xmm0
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %or_from_memory(f32x4, i64) -> f32x4 {
+block0(v0: f32x4, v1: i64):
+    v2 = load.f32x4 notrap aligned v1
+    v3 = bor v0, v2
+    return v3
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   movups  0(%rdi), %xmm4
+;   vorps   %xmm0, %xmm4, %xmm0
```

Comment:
I would have expected that in this case the load could have been merged into `vorps`, any idea why that didn't happen?

Reply:
It looks like for loads to be sunk we need a bunch of additional stuff. Which for some reason I had the idea wasn't necessary before.

https://github.com/bytecodealliance/wasmtime/blob/d30ce3192b416010fe3d6fe3a573cad743b7a2b7/cranelift/codegen/src/isa/x64/lower.isle#L3607-L3610

So that probably explains why this particular case didn't get sunk. `bor` doesen't have a special case for sinking loads for floats (but does have it for integers).

Reply:
Ah yeah I think we'll have to add all the `sinkable_load` business to the rules for all the lowerings here. That shouldn't be too too bad as it's basically the same as all the other arithmetic instructions.